src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -620,6 +630,16 @@ List<StudentParticipation> findParticipationsByStudentIdAndIndividualExercisesWi
     List<StudentParticipation> findTestRunParticipationsByStudentIdAndIndividualExercisesWithEagerSubmissionsResult(@Param("studentId") Long studentId,
             @Param("exercises") List<Exercise> exercises);
 
+    @Query("""
+            SELECT DISTINCT p FROM StudentParticipation p
+            LEFT JOIN FETCH p.submissions s
+            WHERE p.testRun = true
+                AND p.student.id = :#{#studentId}
+                AND p.exercise in :#{#exercises}
+            """)
+    List<StudentParticipation> findTestRunParticipationsByStudentIdAndIndividualExercisesWithEagerSubmissions(@Param("studentId") Long studentId,
+            @Param("exercises") List<Exercise> exercises);
```

Comment:
Why do we potentially duplicate these methods instead of adding a parameter that determines if we should load test tuns or normal participations?

Reply:
I'm not sure. I've just went with the style that was applied so far in this file. I'm sure we can refactor this in a PR scoped to that problem as we would ideally wipe out all 8 places in this file where it's done like that